### ISSUE TITLE
[Question] How to update the gradient of vit?

### ISSUE THREAD

**Issue by Lin-Tianwei** [haotian-liu/LLaVA#1449]
### Question

When I set vit's attribute requires_grad to True, when loss performs gradient backpass, vit's grad is still None. I think this is because the function truncates the gradient postback when organizing text and images as input. Can someone tell me how to update the gradient of vit? Thank you!

**Comment by MrPetrichor** [comment]
I have the same question, after set ViT parameters grad  to True and the parameters  weren't updated : (

**Comment by Lin-Tianwei** [comment]
> I have the same question, after set ViT parameters grad to True and the parameters weren't updated : (

I have solved it! You can delete the decorator **decorator @torch.no_grad()** of **forward** method in **clip_encoder.py**.
